src/main/scala/com/gu/contentapi/services/PageLookup.scala

```diff
@@ -0,0 +1,58 @@
+package com.gu.contentapi.services
+
+import com.gu.contentapi.client._
+import com.gu.contentapi.client.model.v1.SearchResponse
+import com.gu.contentapi.client.model.SearchQuery
+import com.typesafe.scalalogging.StrictLogging
+import scala.concurrent.duration._
+import scala.concurrent.{ Await, Future }
+import com.gu.contentapi.Config.capiKey
+import scala.collection.concurrent.TrieMap
+
+object PageLookup extends StrictLogging {
+
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  sealed trait ResponseFromCapiQuery
+  case class SuccessfulQuery(searchResponse: SearchResponse) extends ResponseFromCapiQuery
+  case class FailedQuery(errorMsg: String) extends ResponseFromCapiQuery
+
+  val client = new GuardianContentClient(capiKey)
+
+  val cache = TrieMap[String, String]()
```

Comment:
You could have used a more featured (you can control size, retention and eviction) and [more performant](https://github.com/ben-manes/caffeine/wiki/Benchmarks)  cache library like [cafeine](https://github.com/blemale/scaffeine) but for the use case your unbounded map should work well!